session/bootstrap.go

```diff
@@ -1190,6 +1193,16 @@ func upgradeToVer52(s Session, ver int64) {
 	doReentrantDDL(s, "ALTER TABLE mysql.stats_histograms MODIFY cm_sketch BLOB(6291456)")
 }
 
+func upgradeToVer53(s Session, ver int64) {
```

Comment:
What will happen if we don't insert this record?

Reply:
I just follow `tidb_enable_window_function`. Actually I don't know what will happen if we don't insert it.

Reply:
I think the default value of `tidb_enable_strict_double_type_check` will be false on an older tidb-server(after upgrading).

Reply:
@bb7133 there has a question, when user upgraded from old version, we should enable strict double type check as default or disable it?

Reply:
I think it's reasonable to keep the 'upgraded cluster' strict check disabled.

> What will happen if we don't insert this record?

Of course there will be no record in for `tidb_enable_strict_double_type_check` in `mysql`.`GLOBAL_VARIABLES`. And the system variable(session/global `tidb_enable_strict_double_type_check`) will use the default value, that is `True`.

So, I will recommend writing a `0` here, which is a disabled strict check for the upgrade cluster, but default value for the brand new cluster. 

@xhebox @tangenta @blacktear23

Reply:
In my opinion, `double(32)` is a real syntax error, that is what MySQL do. So if upgrade to new version, TiDB should prevent user write the SQL like that. And then we provide the `tidb_enable_strict_double_type_check` for legacy system.

Reply:
TiDB set variables in the system table for this purpose:
eg: 
In the rolling update stage, different versions of TiDB exist in the same cluster, if one allows column type change and build a job, once the job is delivered to a non-allowed node, the background DDL owner will get something inconsistent (missing job parameter and can't analyze) and fail.

for this case
The job parameter is still the same, if one allows creating double(10) and build a DDL job, and this job delivered a non-allowed node, the double(10) is still not restricted in the DDL owner handling logic of create-table-action and create success. so is meaningless to set it to 1 in rolling update stage.

Reply:
Gotcha.